packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -78,15 +83,29 @@ class DevFSEntry {
   Future<List<int>> contentsAsBytes() async {
     if (_isSourceEntry)
       return bundleEntry.contentsAsBytes();
-    return file.readAsBytes();
+    if (file is Link) {
```

Comment:
You could refactor this into `_fileFromLink` or something similar that either follows the symlink or returns the file so that you can use it in both places. (where you call `readAsBytes ` and `openRead `)